config/input-resource-verify.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project name="input-validate" default="execute">
+
+  <condition property="checkstyle.ant.skip">
+    <isset property="checkstyle.ant.skip" />
+  </condition>
+
+  <target name="execute" unless="checkstyle.ant.skip">
+    <taskdef name="checkstyle"
+             classname="com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask"
+             classpath="${mvn.runtime_classpath}"
+             />
+
+    <property name="checkstyle.pattern.todo" value="NOTHingWillMatCH_-"/>
+    <property name="check.config" location="config/checkstyle_resource_checks.xml"/>
```

Comment:
@kukreja-vikramaditya After talking with @romani , he agrees about extending this for more checks in the future.
`OuterTypeFilenameCheck` requires `TreeWalker` which we can't run on non-compilable files, so we will have to examine this in new issue outside this PR.

For now, we will make a generic name for file so it makes sense if and when we extend it.
Config file will be named `checkstyle_non_main_files_checks.xml`. All other names required (suppressions and cachefile) should also be named around the config name.